src/thunderbots/software/util/parameter/dynamic_parameters.cpp

```diff
@@ -0,0 +1,32 @@
+#include "util/parameter/dynamic_parameters.h"
+
+namespace DynamicParameters
+{
+void updateAllParametersFromROSParameterServer()
+{
+    Parameter<bool>::updateAllParametersFromROSParameterServer();
+    Parameter<int32_t>::updateAllParametersFromROSParameterServer();
+    Parameter<double>::updateAllParametersFromROSParameterServer();
+    Parameter<std::string>::updateAllParametersFromROSParameterServer();
+    Parameter<std::vector<bool>>::updateAllParametersFromROSParameterServer();
+    Parameter<std::vector<int32_t>>::updateAllParametersFromROSParameterServer();
+    Parameter<std::vector<double>>::updateAllParametersFromROSParameterServer();
+    Parameter<std::vector<std::string>>::updateAllParametersFromROSParameterServer();
+}
+
+const extern Parameter<int> param1("/thunderbots/parameters/param1", 7);
+
+namespace Navigator
+{
+const Parameter<double> param2("/thunderbots/parameters/param2", 55.0);
+}
+
+namespace HL
+{
+namespace STP
+{
+const Parameter<std::vector<std::string>> param3("/thunderbots/parameters/param3",
+                                                 {"MovePlay", "IdlePlay"});
+}
+}
+}
```

Comment:
Just a general note, I think not indenting namespaces (something enforced by our clang formatting) really hurts the readability of files like these. 

There are a few clang options like `CompactNamespaces` and `NamespaceIndentation` that could help (https://clang.llvm.org/docs/ClangFormatStyleOptions.html). @henrymliu @MathewMacDougall what do you think?

Reply:
Agree that the current style is not easy to read in this case. I think
`NamespaceIndentation: NI_ALL`
`CompactNamespaces: false`
would look best. 

The only think to be careful of is that since we have 90 character max width, if we nest too many namespaces our code will get collapsed/formatted oddly, or broken up so much it's also harder to read. Not something I expect, just a note. @henrymliu thought? I can do whatever we decide tonight.

Reply:
Agreed.